### DISCUSSION TITLE
Objects detection parameters linked to zones

### DISCUSSION THREAD

**Discussion by uscan27** [blakeblackshear/frigate#11880] [Ideas]
Hello,

It would be a game changer for me to be able to define objects detection parameters depending on a zone. I want to detect human presence. When a human is far away it's very little on the screen, when he's narrow he's bigger. This could exclude almost all the false positive i get.

Something like that : 

    zones:
      centre:
        coordinates: 639,238,570,250,562,301,541,292,540,264,534,242,524,211,518,165,766,182,714,204,688,232
        objects:
          - person
              min_area: 1000
              max_area: 15000
              threshold: 0.72
              min_score: 0.6
          - dog
          - cat
          - car

Thanks for Frigate, it's awesome :)

**Comment by NickM-27** [comment]
first of all, you can already do this, zones have a `filters` field.

second of all, it is not that simple. From a camera perspective objects often are in a very similar position as they approach the camera they just appear bigger. This type of filtering can also cause gaps in logic, like tuning for adults but then children get filtered out.

**Reply by uscan27** [reply]
Yes it's true, but as far as i know this filter field works for the whole picture, the point here is to constrain it into a define zone.

For the second point, for me, and perhaps others, it would be useful anyway. Frigate detects human presence a lot more tiny than children... because this camera can see far enough for a real human to be that size in a certain zone of my picture. And it works detecting human that small, so I wouldn't like to disable it, just say : in this zone, a human cannot be smaller that this, in this zone a human cannot be bigget that this. This woudl erase a lot of false positives in my experience.

**Reply by NickM-27** [reply]
> Yes it's true, but as far as i know this filter field works for the whole picture, the point here is to constrain it into a define zone.

no, the filter field can be applied on the zone level as well

**Reply by uscan27** [reply]
I tried and it seems to be accepted. Great!!!! I'm going to test it... Thank you
